From 2b49fb312ccf5aedc858e785f65a7046341540a6 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei.xu@nxp.com>
Date: Tue, 5 Aug 2025 17:19:55 +0800
Subject: [PATCH 50/56] LF-15972: arm64: dts: frdm-imx95: enable M.2 Key-E
 power

Enable M.2 Key-E power to make sure it can work on FRDM-IMX95 platform.

Signed-off-by: Lei Xu <lei.xu@nxp.com>
Signed-off-by: Steven Yang <zhao.yang_1@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
Acked-by: Jason Liu <jason.hui.liu@nxp.com>
---
 .../boot/dts/freescale/imx95-15x15-frdm.dts    | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx95-15x15-frdm.dts b/arch/arm64/boot/dts/freescale/imx95-15x15-frdm.dts
index 2c7080404d08..230fb4ed605b 100644
--- a/arch/arm64/boot/dts/freescale/imx95-15x15-frdm.dts
+++ b/arch/arm64/boot/dts/freescale/imx95-15x15-frdm.dts
@@ -205,6 +205,7 @@ reg_usdhc3_vmmc: regulator-usdhc3 {
 		regulator-name = "WLAN_EN";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
+		vin-supply = <&reg_m2_ekey_pwr>;
 		gpio = <&pcal6524 9 GPIO_ACTIVE_HIGH>;
 		/*
 		 * IW612 wifi chip needs more delay than other wifi chips to complete
@@ -223,6 +224,17 @@ reg_vref_1v8: regulator-adc-vref {
 		regulator-max-microvolt = <1800000>;
 	};
 
+	reg_m2_ekey_pwr: regulator-m2-pwr {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ekey_pwr>;
+		regulator-name = "M.2-power-ekey";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio5 11 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
 	reg_m2_mkey_pwr: regulator-m2-mkey-pwr {
 		compatible = "regulator-fixed";
 		pinctrl-names = "default";
@@ -857,6 +869,12 @@ IMX95_PAD_XSPI1_DATA6__GPIO5_IO_BIT6			0x31e
 		>;
 	};
 
+	pinctrl_ekey_pwr: ekeypwrgrp {
+		fsl,pins = <
+			IMX95_PAD_XSPI1_SS1_B__GPIO5_IO_BIT11			0x31e
+		>;
+	};
+
 	pinctrl_mkey_pwr: mkeypwrgrp {
 		fsl,pins = <
 			IMX95_PAD_XSPI1_SS0_B__GPIO5_IO_BIT10			0x31e
-- 
2.34.1

